File: app/src/main/java/knf/kuma/jobscheduler/UpdateJob.java
package knf.kuma.jobscheduler;

import android.app.Notification;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.v4.app.NotificationCompat;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobManager;
import com.evernote.android.job.JobRequest;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.util.concurrent.TimeUnit;

import knf.kuma.Main;
import knf.kuma.R;

public class UpdateJob extends Job {
    public static final String TAG = "update-job";
    public static final String CHANNEL = "app-updater";

    public static void schedule() {
        if (JobManager.instance().getAllJobRequestsForTag(TAG).size() == 0)
            new JobRequest.Builder(TAG)
                    .setPeriodic(TimeUnit.HOURS.toMillis(6))
                    .setRequiredNetworkType(JobRequest.NetworkType.CONNECTED)
                    .setRequirementsEnforced(true)
                    .build().schedule();
    }

    @NonNull
    @Override
    protected Result onRunJob(@NonNull Params params) {
        try {
            Document document = Jsoup.connect("https://raw.githubusercontent.com/jordyamc/UKIKU/master/version.num").get();
            int n_code = Integer.parseInt(document.select("body").first().ownText().trim());
            int s_code = PreferenceManager.getDefaultSharedPreferences(getContext()).getInt("last_notified_update", 0);
            if (n_code <= s_code)
                return Result.SUCCESS;
            int o_code = getContext().getPackageManager().getPackageInfo(getContext().getPackageName(), 0).versionCode;
            if (n_code > o_code) {
                showNotification();
                PreferenceManager.getDefaultSharedPreferences(getContext()).edit().putInt("last_notified_update", n_code).apply();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.SUCCESS;
    }

    private void showNotification() {
        try {
            Notification notification = new NotificationCompat.Builder(getContext(), CHANNEL)
                    .setSmallIcon(R.drawable.ic_not_update)
                    .setContentTitle("UKIKU")
                    .setContentText("Nueva versión disponible")
                    .setContentIntent(PendingIntent.getActivity(getContext(), 5598, new Intent(getContext(), Main.class), PendingIntent.FLAG_UPDATE_CURRENT))
                    .setColor(getContext().getResources().getColor(R.color.colorAccent))
                    .build();
            ((NotificationManager) getContext().getSystemService(Context.NOTIFICATION_SERVICE)).notify(954857, notification);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/knf/kuma/jobscheduler/UpdateJob.kt
package knf.kuma.jobscheduler

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.preference.PreferenceManager
import androidx.core.app.NotificationCompat
import androidx.core.content.ContextCompat
import com.evernote.android.job.Job
import com.evernote.android.job.JobManager
import com.evernote.android.job.JobRequest
import knf.kuma.Main
import knf.kuma.R
import org.jsoup.Jsoup
import java.util.concurrent.TimeUnit

class UpdateJob : Job() {

    override fun onRunJob(params: Job.Params): Job.Result {
        try {
            val document = Jsoup.connect("https://raw.githubusercontent.com/jordyamc/UKIKU/master/version.num").get()
            val nCode = Integer.parseInt(document.select("body").first().ownText().trim { it <= ' ' })
            val sCode = PreferenceManager.getDefaultSharedPreferences(context).getInt("last_notified_update", 0)
            if (nCode <= sCode)
                return Job.Result.SUCCESS
            val oCode = context.packageManager.getPackageInfo(context.packageName, 0).versionCode
            if (nCode > oCode) {
                showNotification()
                PreferenceManager.getDefaultSharedPreferences(context).edit().putInt("last_notified_update", nCode).apply()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return Job.Result.SUCCESS
    }

    private fun showNotification() {
        try {
            val notification = NotificationCompat.Builder(context, CHANNEL)
                    .setSmallIcon(R.drawable.ic_not_update)
                    .setContentTitle("UKIKU")
                    .setContentText("Nueva versión disponible")
                    .setContentIntent(PendingIntent.getActivity(context, 5598, Intent(context, Main::class.java), PendingIntent.FLAG_UPDATE_CURRENT))
                    .setColor(ContextCompat.getColor(context, R.color.colorAccent))
                    .build()
            (context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager).notify(954857, notification)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    companion object {
        const val TAG = "update-job"
        const val CHANNEL = "app-updater"

        fun schedule() {
            if (JobManager.instance().getAllJobRequestsForTag(TAG).size == 0)
                JobRequest.Builder(TAG)
                        .setPeriodic(TimeUnit.HOURS.toMillis(6))
                        .setRequiredNetworkType(JobRequest.NetworkType.CONNECTED)
                        .setRequirementsEnforced(true)
                        .build().schedule()
        }
    }
}
